hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -997,6 +998,15 @@ private void addNotificationLog(NotificationEvent event, ListenerEvent listenerE
         stmt.execute("SET @@session.sql_mode=ANSI_QUOTES");
       }
 
+      // Derby doesn't allow FOR UPDATE to lock the row being selected (See https://db.apache
```

Comment:
For addNotificationLog ..the lock is already taken by the methods in txnHandler class. addNotificationLog is always and should be called for events generated by txnHandler class ..so i think this lock is not required.

Reply:
The callers of ObjectStore#addNotificationEvents() will not take derbyLock and thus they will be able to access NOTIFICATION_SEQUENCE while the txnHandler callers are accessing it. We need same synchronization in both the methods.